experiments/Compute.Experiments/Az.Compute.psm1

```diff
@@ -0,0 +1,331 @@
+    )
+
+    $pip = Get-AzureRmPublicIpAddress | Where-Object { $_.Name -eq $Name } | Select-Object -First 1 -Wait;
+
+    if($pip -eq $null) {
+        # Create a public IP address and specify a DNS name.
+        return New-AzureRmPublicIpAddress `
+            -ResourceGroupName $ResourceGroup `
+            -Location $Location `
+            -Name $Name `
+            -DomainNameLabel $DnsLabel `
+            -AllocationMethod $AllocationMethod `
+            -IdleTimeoutInMinutes $IdleTimeoutInMinutes;
+    } else {
+        return $pip;
+    }
+}
+
+function Use-Nsg {
+    param (
+        [Parameter(Mandatory=$true)] [string] $Name,
+        [Parameter(Mandatory=$true)] [string] $ResourceGroup,
+        [Parameter(Mandatory=$true)] [string] $Location,
+        [Parameter(Mandatory=$true)] [int[]] $OpenPorts
+    )
+
+    $nsg = Get-AzureRmNetworkSecurityGroup | Where-Object { $_.Name -eq $Name } | Select-Object -First 1 -Wait;
+
+    if($nsg -eq $null) {
+        $rules = New-Object `
+            "System.Collections.Generic.List[Microsoft.Azure.Commands.Network.Models.PSSecurityRule]";
+        $priority = 1000;
+
+        foreach($port in $OpenPorts)
+        {
+            $nsgRule = New-AzureRmNetworkSecurityRuleConfig `
+                -Name myNetworkSecurityGroupRuleRDP `
+                -Protocol Tcp `
+                -Direction Inbound `
+                -Priority $priority `
+                -SourceAddressPrefix * `
+                -SourcePortRange * `
+                -DestinationAddressPrefix * `
+                -DestinationPortRange $port `
+                -Access Allow;
+            $rules.Add($nsgRule);
+
+            $priority--;
+        }
+
+        # Create an NSG.
+        return New-AzureRmNetworkSecurityGroup `
+            -ResourceGroupName $ResourceGroup `
+            -Location $Location `
+            -Name $Name `
+            -SecurityRules $rules;
+    } else {
+        return $nsg;
+    }
+}
+
+function Use-Nic {
+    param (
+        [Parameter(Mandatory=$true)] [string] $Name,
+        [Parameter(Mandatory=$true)] [string] $ResourceGroup,
+        [Parameter(Mandatory=$true)] [string] $Location,
+        [Parameter(Mandatory=$true)] [string] $SubnetId,
+        [Parameter(Mandatory=$true)] [string] $PublicIpAddressId,
+        [Parameter(Mandatory=$true)] [psobject] $NetworkSecurityGroupId
+    )
+
+    $nic = Get-AzureRmNetworkInterface | Where-Object { $_.Name -eq $Name } | Select-Object -First 1 -Wait;
+
+    if($nic -eq $null) {
+        # Create a virtual network card and associate with public IP address and NSG
+        return New-AzureRmNetworkInterface `
+            -Name $Name `
+            -ResourceGroupName $resourceGroup `
+            -Location $location `
+            -SubnetId $SubnetId `
+            -PublicIpAddressId $PublicIpAddressId `
+            -NetworkSecurityGroupId $NetworkSecurityGroupId.ToString();
+    } else {
+        return $nic;
+    }
+}
+
+function Write-Info {
```

Comment:
If you implement a cmdlet, you will have access to Write-Verbose, Write-Information, and Write-Proggress